contracts/liquidity_hub/pool-manager/src/tests/integration_tests.rs

```diff
@@ -2649,3 +2650,436 @@ mod locking_lp {
+            },
+            burn_fee: Fee {
+                share: Decimal::zero(),
+            },
+            osmosis_fee: Fee {
+                share: Decimal::zero(),
+            },
+            extra_fees: vec![],
+        };
+
+        // Create a pair
+        suite.instantiate_default().create_pair(
+            creator.clone(),
+            asset_denoms,
+            vec![6u8, 6u8],
+            pool_fees,
+            white_whale_std::pool_network::asset::PairType::ConstantProduct,
+            Some("whale-uluna".to_string()),
+            vec![coin(1000, "uusd")],
+            |result| {
+                result.unwrap();
+            },
+        );
+
+        let contract_addr = suite.pool_manager_addr.clone();
+        let incentive_manager_addr = suite.incentive_manager_addr.clone();
+        let lp_denom = suite.get_lp_denom("whale-uluna".to_string());
+
+        // Lets try to add liquidity
+        suite
+            .provide_liquidity(
+                creator.clone(),
+                "whale-uluna".to_string(),
+                None,
+                None,
+                vec![],
+                |result| {
+                    let err = result.unwrap_err().downcast::<ContractError>().unwrap();
+
+                    match err {
+                        ContractError::EmptyAssets => {}
+                        _ => panic!("Wrong error type, should return ContractError::EmptyAssets"),
+                    }
+                },
+            )
+            .provide_liquidity(
+                creator.clone(),
+                "whale-uluna".to_string(),
+                None,
+                None,
+                vec![Coin {
+                    denom: "uosmo".to_string(),
+                    amount: Uint128::from(1_000_000u128),
+                }],
+                |result| {
+                    let err = result.unwrap_err().downcast::<ContractError>().unwrap();
+
+                    match err {
+                        ContractError::AssetMismatch { .. } => {}
+                        _ => panic!("Wrong error type, should return ContractError::AssetMismatch"),
+                    }
+                },
+            )
+            .provide_liquidity(
+                creator.clone(),
+                "whale-uluna".to_string(),
+                None,
+                None,
+                vec![Coin {
+                    denom: "uwhale".to_string(),
+                    amount: Uint128::from(1_000_000u128),
+                }],
+                |result| {
+                    let err = result.unwrap_err().downcast::<ContractError>().unwrap();
+
+                    match err {
+                        ContractError::InvalidZeroAmount {} => {}
+                        _ => panic!(
+                            "Wrong error type, should return ContractError::InvalidZeroAmount"
+                        ),
+                    }
+                },
+            )
+            .provide_liquidity(
```

Comment:
Seems duplicated

Reply:
kinda but not, the order or the assets is inverted but yeah